### ISSUE TITLE
Dataset with the labels of a rectangle

### ISSUE THREAD

**Issue by PROCESS-Y** [wgcban/ChangeFormer#22]
Hello, thank you very much for your work. If my dataset only has the labels of a rectangle, how should I run the code?
(⊙o⊙)…

**Comment by wgcban** [comment]
Hi @PROCESS-Y 
If you have rectangular shaped images (i.e., w!=h), I would suggest you to do either one of the following:

1. During training you can use a random crop of size 256x256 from the rectangular-shaped images
2. Crop images of size 256x256 from the rectangle-shaped images to form a new dataset

Please let me know if you need more help.

**Comment by PROCESS-Y** [comment]
Sorry, I didn't make myself clear. My problem is that my dataset only has labels of bounding box. What I'm doing is defect detection of physical images of PCB based on CAD. I marked the defect location with a bounding box.

**Comment by wgcban** [comment]
Hi @PROCESS-Y 

I see. Now I understood your problem. Basically, you are doing object detection. I think in this case you need a couple of changes to the dataloader and loss functions. Basically, you need to add a bounding box regression layer (read [here](https://towardsdatascience.com/bounding-box-prediction-from-scratch-using-pytorch-a8525da51ddc) at the end of the network to predict bounding boxes and also you should train the network accordingly with the Intersection Over Union metric ([read here](https://engineering.purdue.edu/DeepLearn/pdf-kak/Detection.pdf)).

Or you can go with simple object detection codes. I have attached couple of resources here: 
[A Tutorial to Object Detection](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Object-Detection)
[RetinaNet](https://github.com/yhenon/pytorch-retinanet)

I hope this answers your question. 

Best

Chaminda

**Comment by PROCESS-Y** [comment]
Thank you for your patient help and actionable advice. Your work has inspired me to address the differences between PCB and CAD. I will try to use this useful information to solve my problem.
 (๑╹ヮ╹๑)ﾉ Thanks ♪

**Comment by wgcban** [comment]
You are welcome @PROCESS-Y.
Actually, that is a really nice application of change detection. 
Good luck with your work!

-Chaminda
